Core/src/org/sleuthkit/autopsy/datasourceprocessors/xry/XRYDataSourceProcessor.java

```diff
@@ -0,0 +1,242 @@
+ *
+ * Copyright 2019 Basis Technology Corp.
+ * Contact: carrier <at> sleuthkit <dot> org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sleuthkit.autopsy.datasourceprocessors.xry;
+
+import com.google.common.collect.Lists;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+import java.util.logging.Level;
+import java.util.stream.Collectors;
+import javax.swing.JPanel;
+import javax.swing.SwingWorker;
+import org.openide.util.NbBundle;
+import org.openide.util.lookup.ServiceProvider;
+import org.openide.util.lookup.ServiceProviders;
+import org.sleuthkit.autopsy.casemodule.Case;
+import org.sleuthkit.autopsy.casemodule.NoCurrentCaseException;
+import org.sleuthkit.autopsy.casemodule.services.FileManager;
+import org.sleuthkit.autopsy.corecomponentinterfaces.DataSourceProcessor;
+import org.sleuthkit.autopsy.corecomponentinterfaces.DataSourceProcessorCallback;
+import org.sleuthkit.autopsy.corecomponentinterfaces.DataSourceProcessorProgressMonitor;
+import org.sleuthkit.autopsy.coreutils.Logger;
+import org.sleuthkit.datamodel.AbstractFile;
+import org.sleuthkit.datamodel.LocalFilesDataSource;
+import org.sleuthkit.datamodel.TskCoreException;
+import org.sleuthkit.datamodel.TskDataException;
+
+/**
+ * An XRY Report data source processor.
+ */
+@ServiceProviders(value = {
+    @ServiceProvider(service = DataSourceProcessor.class)}
+)
+public class XRYDataSourceProcessor implements DataSourceProcessor {
+
+    private final XRYDataSourceProcessorConfigPanel configPanel;
+    
+    //Background processor to relieve the EDT from adding files to the case
+    //database and parsing the report files.
+    private XRYReportProcessorSwingWorker swingWorker;
+
+    private static final Logger logger = Logger.getLogger(XRYDataSourceProcessor.class.getName());
+
+    public XRYDataSourceProcessor() {
+        configPanel = XRYDataSourceProcessorConfigPanel.getInstance();
+    }
+
+    @Override
+    @NbBundle.Messages({
+        "XRYDataSourceProcessor.dataSourceType=Import Tool Report"
+    })
+    public String getDataSourceType() {
+        return Bundle.XRYDataSourceProcessor_dataSourceType();
+    }
+
+    @Override
+    public JPanel getPanel() {
+        return configPanel;
+    }
+
+    @Override
+    public boolean isPanelValid() {
+        return true;
```

Comment:
Shouldn't we do the validation that occurs in the run method here? It seems that at a minimum, we need to be sure that some folder path has been entered in the text field.

Reply:
Okay, is there a way to pass along messages for why it is not a valid panel?

Reply:
I'd take a look at what the other DSPs do.

If you address the other little comments, I will take this PR and we can send this to field trial without the more nuanced error handling.

Reply:
I see. I was able to move it and still show the messages.